### ISSUE TITLE
Typerror on Php 8.2

### ISSUE THREAD

**Issue by Whip** [composer/composer#12156]
My `composer.json`:

```json
{
    "require": {
        "phpmailer/phpmailer": "^6.9",
        "google/cloud-recaptcha-enterprise": "^1.13",
        "google/apiclient": "^2.17",
        "kreait/firebase-php": "^6.9",
        "lbuchs/webauthn": "^2.2"
    },
    "autoload": {
        "psr-4": {
            "GStone\\": "src/"
        }
    }
}

```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: OK
Checking git settings: No git process found
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking Composer version: OK
Checking Composer and its dependencies for vulnerabilities: OK
Composer version: 2.8.1
PHP version: 8.2.12
PHP binary path: C:\xampp\php\php.exe
OpenSSL version: OpenSSL 3.0.11 19 Sep 2023
curl version: 8.4.0 libz 1.2.12 ssl OpenSSL/3.0.11
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
zip: extension present, unzip not available, 7-Zip not available
```

When I run this command:

```
composer update
```

I get the following output:

```
In Problem.php line 652:

  [TypeError]
  substr(): Argument #1 ($string) must be of type string, null given


Exception trace:
  at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Problem.php:652
 substr() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Problem.php:652
 Composer\DependencyResolver\Problem::Composer\DependencyResolver\{closure}() at n/a:n/a
 array_map() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Problem.php:651
 Composer\DependencyResolver\Problem::getProvidersList() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Problem.php:281
 Composer\DependencyResolver\Problem::getMissingPackageReason() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Rule.php:324
 Composer\DependencyResolver\Rule->getPrettyString() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Problem.php:168
 Composer\DependencyResolver\Problem::formatDeduplicatedRules() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/Problem.php:111
 Composer\DependencyResolver\Problem->getPrettyString() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/DependencyResolver/SolverProblemsException.php:52
 Composer\DependencyResolver\SolverProblemsException->getPrettyString() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Installer.php:513
 Composer\Installer->doUpdate() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Installer.php:298
 Composer\Installer->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Command/UpdateCommand.php:281
 Composer\Command\UpdateCommand->execute() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:1040
 Symfony\Component\Console\Application->doRunCommand() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:301
 Symfony\Component\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:407
 Composer\Console\Application->doRun() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/src/Composer/Console/Application.php:147
 Composer\Console\Application->run() at phar://C:/ProgramData/ComposerSetup/bin/composer.phar/bin/composer:93
 require() at C:\ProgramData\ComposerSetup\bin\composer.phar:29

update [--with WITH] [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-audit] [--audit-format AUDIT-FORMAT] [--no-autoloader] [--no-suggest] [--no-progress] [-w|--with-dependencies] [-W|--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-m|--minimal-changes] [--patch-only] [-i|--interactive] [--root-reqs] [--bump-after-update [BUMP-AFTER-UPDATE]] [--] [<packages>...]
```
I have reset my PC and doing a fresh install of everything. This is running on a Windows 11 computer with Xampp.
Could this have something to do with unzip? The unzip extension is enabled in php.ini.

**Comment by baptistajh** [comment]
Same problem then #12151 already have a fix PR open and reviewed just waiting for aproval #12152

**Comment by Seldaek** [comment]
Fixed by #12152
